### PR TITLE
Feat/fxa

### PR DESCRIPTION
**Pull request by alsprogrammer**

### CODE REVIEW

File: test/frontend/tests/workflow-tests.js [hunk 1]
```diff
@@ -18,87 +20,61 @@ navigator.serviceWorker.register('/serviceWorker.js');
...
+        password: 'magic'
+      },
+      correctToken
+    );
     await fr.getMetadata();
     await fr.download(options);
     assert.equal(fr.state, 'complete');
   });
 
-  it('blocks invalid passwords from downloading', async function() {
```
Comment (aesariyev) [inline test/frontend/tests/workflow-tests.js:58]: Can we rework this with the invalid token?

File: test/frontend/tests/workflow-tests.js [hunk 2]
```diff
@@ -18,87 +20,61 @@ navigator.serviceWorker.register('/serviceWorker.js');
 
 describe('Upload / Download flow', function() {
   this.timeout(0);
-  it('can only download once by default', async function() {
```
Comment (aesariyev) [inline test/frontend/tests/workflow-tests.js:21]: Can we keep the original logic? It should throw an error when download was performed previously

File: server/fxa-mock.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+module.exports = {
+  token: correct_token,
+  getFxaConfig,
+  verify: async function(token) {
+    if (!token) {
+      return null;
+    }
+
+    if (token == correct_token) {
+      return {
```
Comment (aesariyev) [inline server/fxa-mock.js:46]: nit: for JSON test inputs we can use fixtures

**Review by aesariyev — CHANGES_REQUESTED** [review]

File: test/frontend/tests/workflow-tests.js (hunk 2 above)
Comment (alsprogrammer) [reply test/frontend/tests/workflow-tests.js:21]: Our default behavior would be to keep files until they are deleted. But yes, good point, thanks, restored.

File: test/frontend/tests/workflow-tests.js (hunk 1 above)
Comment (alsprogrammer) [reply test/frontend/tests/workflow-tests.js:58]: The problem here is that either the bearer token or the password are passed in the same way: as the `Authenticaion` header. 

In the previous version the `hmac` auth method was used, which required the password. To implement it, we have to pass the password in a different way (probably, as a different header?). 

Should I implement it?

File: server/fxa-mock.js (hunk 3 above)
Comment (alsprogrammer) [reply server/fxa-mock.js:46]: As per my understanding, the fixtures are used to make the code clearer (for exmple, hiding the auxilary code). 

Not sure if it make the code clearer here, as we can see exactly what's returned.

WDYT?

File: test/frontend/tests/workflow-tests.js (hunk 1 above)
Comment (aesariyev) [reply test/frontend/tests/workflow-tests.js:58]: I thought we can add test case for the token with invalid issuer or expiration date, so the test will throw an expected exception

File: server/fxa-mock.js (hunk 3 above)
Comment (aesariyev) [reply server/fxa-mock.js:46]: Yep, this comment doesn't affect on the review decision overall. Just an observation

**Review by aesariyev — APPROVED** [review]

**Pull request merged**